core/common/src/test/java/alluxio/security/authentication/GrpcSecurityTest.java

```diff
@@ -120,7 +121,7 @@ public void testAuthMismatch() throws Exception {
     mConfiguration.set(PropertyKey.SECURITY_AUTHENTICATION_TYPE, AuthType.SIMPLE);
     GrpcChannelBuilder channelBuilder =
         GrpcChannelBuilder.newBuilder(getServerConnectAddress(server), mConfiguration);
-    mThrown.expect(UnauthenticatedException.class);
+    mThrown.expect(UnimplementedException.class);
```

Comment:
Why does this throw `Unimplemented`?

Reply:
Nice catch. So technically it's a notimplemented exception because the server in this test don't even register the authentication service. I think unimplemented fits better for this case. However we could blanket notimplemented with unauthenticated if necessary.

Reply:
Unauthenticated might be better? But when I run this test now the error seems kind of confusing. It doesn't necessarily indicate that the security configuration is incorrect. The current message is:

```
alluxio.exception.status.UnimplementedException: Method not found: alluxio.grpc.sasl.SaslAuthenticationService/authenticate
```

Would it be possible in this case to either 1) improve the message to include some kind of auth mistmatch 2) create a new exception class which represents authentication mismatching. Right now the error seems crytpic.